src/app/main/subsysmgr.c

```diff
@@ -230,6 +230,27 @@ subsystems_postfork(void)
   }
 }
 
+/**
+ * Run thread-init code on all subsystems that declare any
+ **/
+void
+subsystems_thread_init(void)
+{
+  check_and_setup();
+
+  for (int i = (int)n_tor_subsystems - 1; i >= 0; --i) {
```

Comment:
Do we want to start at the lowest subsystem first?

Reply:
Oh no... Good catch, that's definitely a copy-paste error :( I'll wait and see if any other mistakes are found, and I'll fix this to follow `subsystems_init_upto`, so `for (unsigned i = 0; i < n_tor_subsystems; ++i)`. Thanks!